reana-ui/src/pages/workflowList/components/WorkflowStatusFilter.js

```diff
@@ -2,39 +2,94 @@
   -*- coding: utf-8 -*-
 
   This file is part of REANA.
-  Copyright (C) 2020 CERN.
+  Copyright (C) 2020, 2022 CERN.
 
   REANA is free software; you can redistribute it and/or modify it
   under the terms of the MIT License; see LICENSE file for more details.
 */
 
 import PropTypes from "prop-types";
-import { Dropdown } from "semantic-ui-react";
+import { useEffect, useState } from "react";
+import { Checkbox, Dropdown, Grid } from "semantic-ui-react";
 
+import { WORKFLOW_STATUSES } from "~/config";
 import { statusMapping } from "~/util";
 
+function removeDeleted(statuses) {
+  return statuses.filter((status) => status !== "deleted");
+}
+
+const NON_DELETED_STATUSES = removeDeleted(WORKFLOW_STATUSES);
+
 const statusOptions = Object.keys(statusMapping).map((status) => ({
   key: status,
   text: status,
   value: status,
   icon: statusMapping[status].icon,
 }));
 
-export default function WorkflowStatusFilters({ valueList, filter }) {
+const nonDeletedStatusOptions = statusOptions.filter(({ value: status }) =>
+  NON_DELETED_STATUSES.includes(status)
+);
+
+function hasSameElements(x, y) {
+  if (x.length !== y.length) {
+    return false;
+  }
+  for (let element of x) {
+    if (!y.includes(element)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+export default function WorkflowStatusFilters({ statusFilter, filter }) {
+  const [valueList, setValueList] = useState([]);
+  const [showDeleted, setShowDeleted] = useState(true);
+
+  useEffect(() => {
+    // Keep the list of selected statuses in sync with the filter used to make the
+    // requests to the API
+    let selection = valueList;
+    if (!showDeleted) {
+      selection = removeDeleted(selection);
+      if (selection.length === 0) {
+        selection = NON_DELETED_STATUSES;
+      }
+    }
+    if (!hasSameElements(selection, statusFilter)) {
+      filter(selection);
+    }
+  }, [filter, statusFilter, showDeleted, valueList]);
```

Comment:
What do you think if we would simplify this code like this:
```js
  import isEqual from "lodash/isEqual";

  useEffect(() => {
    // Keep the list of selected statuses in sync with the filter used to make the
    // requests to the API
    let selection = [...valueList];
    if (!showDeleted) selection = removeDeleted(selection);
    if (!showDeleted && !selection.length) selection = NON_DELETED_STATUSES;

    if (!isEqual(selection, statusFilter)) {
      filter(selection);
    }
  }, [filter, statusFilter, showDeleted, valueList]);
  ```
  In this way we can get rid of `hasSameElements()` and replace it with `isEqual()` from `lodash`

Reply:
Thank you! I did not know about `isEqual`